File: src/my_project/medium_problems/from101to150/sort_linked_list.py
from typing import List, Optional 

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
#         self.next = next
class Solution:
    def sortList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        if not head:
            return None
        myList=[]
        while head:
            myList.append(head.val)
            head=head.next
        myList.sort()
        newHead=point=ListNode(0)
        for i in range(len(myList)):
            point.next=ListNode(myList[i])
            point=point.next
        return newHead.next
        

File: my_project_typescript/medium_problems_typescript/from51to100/sortLinkedList.ts

class ListNode {
     val: number
     next: ListNode | null
     constructor(val?: number, next?: ListNode | null) {
         this.val = (val===undefined ? 0 : val)
         this.next = (next===undefined ? null : next)
    }
 }


 function sortList(head: ListNode | null): ListNode | null {
     if (!head){
         return null
     }
     let myList: Array<number> = []
     while (head){
        myList.push(head.val)
        head = head.next
     }

     myList.sort( (a,b) => a<=b?-1:1)

     let point: ListNode = new ListNode(0)
     let newHead: ListNode = point

     for (let i = 0; i < myList.length; i++){
         point.next = new ListNode(myList[i])
         point = point.next
     }
    return newHead.next

};